File: high-quality code/high quality classes/homework/Abstraction/Rectangle.cs
using System;

namespace Abstraction
{
    class Rectangle : Figure
    {
        private double width;
        private double height;        

        public Rectangle(double width, double height)
            :base()
        {
            this.Width = width;
            this.Height = height;
        }

        public double Width
        {
            get
            {
                return this.width;
            }
            set
            {
                if (value <= 0)
                {
                    throw new ArgumentOutOfRangeException("Width", "Width cannot be negative or equal to 0.");
                }

                this.width = value;
            }
        }
        public double Height
        {
            get
            {
                return this.height;
            }
            set
            {
                if (value <= 0)
                {
                    throw new ArgumentOutOfRangeException("Height", "Height cannot be negative or equal to 0.");
                }

                this.height = value;
            }
        }
        public override double CalcPerimeter()
        {
            double perimeter = 2 * (this.Width + this.Height);
            return perimeter;
        }

        public override double CalcSurface()
        {
            double surface = this.Width * this.Height;
            return surface;
        }
    }
}

File: OOP/11. OOP in Java/homework/src/geometry/Rectangle.java
package geometry;

public class Rectangle extends PlaneShape {
	private double width;
	private double height;

	public Rectangle(Vertex[] vertices, final double width, final double height) {
		super(vertices);

		this.setHeight(height);
		this.setWidth(width);
	}

	public double getWidth() {
		return width;
	}

	private void setWidth(double width) {
		if (width <= 0) {
			throw new IllegalArgumentException(
					"Width cannot be a negative number.");
		}

		this.width = width;
	}

	public double getHeight() {
		return height;
	}

	private void setHeight(double height) {
		if (height <= 0) {
			throw new IllegalArgumentException(
					"Height cannot be a negative number.");
		}

		this.height = height;
	}

	@Override
	public double getPerimeter() {
		double perimeter = (this.width + this.height) * 2;
		return perimeter;
	}

	@Override
	public double getArea() {
		double area = this.width * this.height;
		return area;
	}

	@Override
	public String toString() {
		return String
				.format("Shape type: %s%nvertex x: %.2f, y: %.2f;%nperimeter: %.2f, area: %.2f",
						this.getClass().getSimpleName(),
						this.getVertex()[0].getX(), this.getVertex()[0].getY(),
						this.getPerimeter(), this.getArea());
	}
}
